qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDefinition.java

```diff
@@ -270,6 +273,37 @@ private CompletableFuture<List<? extends LocationLink>> findDefinitionFromPart(P
 		}
 	}
 
+	/**
+	 * Find parameter definition for Enum in #when section.
+	 *
+	 * @param parameter
+	 * @param template
+	 * @param cancelChecker
+	 *
+	 * @return parameter definition for Enum in #when section.
+	 */
+	private CompletableFuture<List<? extends LocationLink>> findDefinitionFromParameter(Parameter parameter,
+			Template template, CancelChecker cancelChecker) {
+		String projectUri = template.getProjectUri();
+		Section whenSection = parameter.getParentSection().getParentSection();
+		if (whenSection != null && whenSection.getSectionKind() == SectionKind.WHEN) {
+			// Ex: {#when Machine.status}
+			// {#is ON }
+			// {#is OF|F}
+			// {/when}
+			Parameter value = ((BaseWhenSection) whenSection).getValueParameter();

```

Comment:
I think value can be null in the case of `{#switch}`, you need to check if value is null. Please write a test with this usecase.